controllers/watch_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestMachineToInfrastructureMapFunc(t *testing.T) {
+	var testcases = []struct {
+		name    string
+		input   schema.GroupVersionKind
+		request handler.MapObject
+		output  []reconcile.Request
+	}{
+		{
+			name: "reconcile infra-1",
+			input: schema.GroupVersionKind{
+				Group:   "foo.cluster.sigs.k8s.io",
+				Version: "v1alpha2",
+				Kind:    "TestMachine",
+			},
+			request: handler.MapObject{
```

Comment:
It looks like the only thing changed between all of these test cases is the config ref, could we only supply the config ref in the table and build the rest of this inside the subtest to reduce some of this boilerplate that doesn't appear to change?

Reply:
yurp